Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/threshold/ThresholdedRandomCutForest.java

```diff
@@ -361,14 +489,20 @@ public AnomalyDescriptor process(double[] inputPoint, long timestamp) {
             scaledInput = applyImpute(scaledInput, timestamp);
         }
 
-        // the following handles both external and internal shingling
-        double[] point = forest.transformToShingledPoint(scaledInput);
+        AnomalyDescriptor result = null;
+        if (getResult) {
```

Comment:
If this block only needs to executed once after all the updates are in, maybe we can create a separate method for it instead of passing a boolean flag?

Reply:
Execute once only after the initial updates are in. Refactored. The update steps are repeated though (without providing any output).